src/transformers/modeling_utils.py

```diff
@@ -4778,6 +4778,7 @@ def _assign_original_dtype(module):
             model.hf_quantizer = hf_quantizer
 
         if _adapter_model_path is not None:
+            adapter_kwargs.update({"key_mapping": key_mapping})
```

Comment:
```suggestion
            adapter_kwargs["key_mapping"] = key_mapping
```

Note related to this PR, but it might be a good idea to change this line:

https://github.com/huggingface/transformers/blob/ecf6eacf3520cfca7e2e0434d356855513bffcc8/src/transformers/modeling_utils.py#L4245

to `adapter_kwargs = kwargs.pop("adapter_kwargs", {}).copy()` to avoid mutating the original dict.